File: DupplicateWithinKDistance.py
def checkDuplicatesWithinK(arr, n, k):
    # Tạo một set (tập hợp) rỗng để lưu các phần tử đã xuất hiện trong mảng
    myset = set()
    for i in range(n):
        # Kiểm tra xem phần tử hiện tại đã xuất hiện trong tập hợp myset chưa
        if arr[i] in myset:
            # Nếu có, nghĩa là phần tử đã xuất hiện trong phạm vi k chỉ số
            return True        
        # Thêm phần tử hiện tại vào set
        myset.add(arr[i])
        # Nếu chỉ số hiện tại >= k, xóa phần tử nằm ngoài phạm vi k từ tập hợp
        if i >= k:
            myset.remove(arr[i - k])
    # Nếu không tìm thấy phần tử trùng lặp trong phạm vi k, trả về False
    return False

if __name__ == "__main__":
    arr = list(map(int, input("Nhập mảng các phần tử (cách nhau bởi dấu cách): ").split()))
    
    k = int(input("Nhập giá trị k (khoảng cách tối đa): "))
    n = len(arr)
    # Kiểm tra xem có phần tử trùng lặp trong khoảng cách k chỉ số không
    if checkDuplicatesWithinK(arr, n, k):
        print("True")
    else:
        print("False")

File: DupplicateWithinKDistance.c
#include <stdio.h>
#include <stdbool.h>
bool checkDuplicatesWithinK(int arr[], int n, int k) {
    for (int i = 0; i < n; i++) {
        // Kiểm tra tất cả các phần tử trong phạm vi k từ chỉ số hiện tại
        for (int j = i + 1; j <= i + k && j < n; j++) {
            if (arr[i] == arr[j]) {
                return true;
            }
        }
    }
    return false;  // Không tìm thấy phần tử trùng lặp trong phạm vi k
}
int main() {
    int n, k;
    printf("Nhập số lượng phần tử mảng: "); scanf("%d", &n);
    int arr[n];
    printf("Nhập các phần tử của mảng: ");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }
    printf("Nhập giá trị k: "); scanf("%d", &k);
    if (checkDuplicatesWithinK(arr, n, k)) {
        printf("True\n"); 
    } else {
        printf("False\n");
    }
    return 0;
}
